rust/ethers-prometheus/src/middleware/mod.rs

```diff
@@ -361,6 +415,91 @@ impl<M: Middleware> Middleware for PrometheusMiddleware<M> {
 
         Ok(result?)
     }
+
+    async fn get_logs(&self, filter: &Filter) -> Result<Vec<Log>, Self::Error> {
```

Comment:
Do you think it's worth doing a histogram? Ofc would mean way more data being sent to Grafana but might offer more insight than the long-running average that we get here.
Thinking out loud, probably best to ship this and if we ever feel like it's not enough, just add histograms or something